File: evidence/generated-systems/golang/fintech-on-golang/fintech-on-golang/internal/model/Customer.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// Customer Declaration
//==============================================================
type Customer struct {
    gorm.Model
     FirstName                                    string
    LastName                                    string
    DateOfBirth                                                            time.Time
    Email                                                            string
    Phone                                                            string
    Address                                                            string
    TaxId                                                            string
    RiskScore                                                            string
    InstitutionId         *uint
    Institution           *FinancialInstitution `gorm:"foreignKey:InstitutionId"`
     Accounts           []Account `gorm:"foreignKey:AccountsFromCustomerId"`
     Wallets           []Wallet `gorm:"foreignKey:WalletsFromCustomerId"`
     Cards           []PaymentCard `gorm:"foreignKey:CardsFromCustomerId"`
     KycProfiles           []KYCProfile `gorm:"foreignKey:KycProfilesFromCustomerId"`
     Consents           []Consent `gorm:"foreignKey:ConsentsFromCustomerId"`
     Agreements           []Agreement `gorm:"foreignKey:AgreementsFromCustomerId"`
     LoanApplications           []LoanApplication `gorm:"foreignKey:LoanApplicationsFromCustomerId"`
     Loans           []Loan `gorm:"foreignKey:LoansFromCustomerId"`
     Portfolios           []InvestmentPortfolio `gorm:"foreignKey:PortfoliosFromCustomerId"`
     Disputes           []Dispute `gorm:"foreignKey:DisputesFromCustomerId"`
    CustomerType                      CustomerType

// parent associations as their child

}

File: evidence/generated-systems/rubyonrails/fintech-on-rails/fintechonrails/app/models/customer.rb
class Customer < ApplicationRecord
  enum CustomerType: [:Individual, :Business]


  composed_of :email,
    class_name: "Email",
    mapping: [
      %w[email_value value]
    ]

  composed_of :phoneNumber,
    class_name: "PhoneNumber",
    mapping: [
      %w[phoneNumber_value value]
    ]

  composed_of :address,
    class_name: "Address",
    mapping: [
      ${$mapping}, 
      ${$mapping}, 
      ${$mapping}, 
      ${$mapping}, 
      %w[address_country country]
    ]

  composed_of :taxId,
    class_name: "TaxId",
    mapping: [
      %w[taxId_value value]
    ]

  composed_of :riskScore,
    class_name: "RiskScore",
    mapping: [
      %w[riskScore_value value]
    ]

  has_many :Institution, class_name: 'FinancialInstitution'
  has_many :Accounts, class_name: 'Account'
  has_many :Wallets, class_name: 'Wallet'
  has_many :Cards, class_name: 'PaymentCard'
  has_many :KycProfiles, class_name: 'KYCProfile'
  has_many :Consents, class_name: 'Consent'
  has_many :Agreements, class_name: 'Agreement'
  has_many :LoanApplications, class_name: 'LoanApplication'
  has_many :Loans, class_name: 'Loan'
  has_many :Portfolios, class_name: 'InvestmentPortfolio'
  has_many :Disputes, class_name: 'Dispute'

end
